2022/Q3/artifacts/PoS-pseudocode/PoS-model.md

```diff
@@ -304,6 +353,17 @@ func withdraw(validator_address, delegator_address)
         computed_amounts = computed_amounts \ {slashed_amount}
       computed_amounts = computed_amounts \union {SlashedAmount{epoch: slash.epoch, amount: updated_amount*slash.rate}}
 
+      // Regard validator's incoming_redelegations - there might be slashes on 
+      // the source validator that should be applied to this bond
+      forall (<dest, redelegations> in validator[validator_address].incoming_redelegations) do
+        var total_redelegated_and_slashable = 0
+        forall (<owner, redelegation> in redelegations | owner == delegator_address) do
+        // Find any redelegation where slash.epoch is between their start and end epoch
+        if (redelegation.start <= slash.epoch && slash.epoch <= redelegation.end) then
```

Comment:
I wonder if this is correct. What if the `src_validator` misbehaved before `redelegation.start` but the evidence was found after `redelegation.start`? I guess this is possible and we should slash.